manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -40,6 +40,12 @@ type TaskReaper struct {
 	cleanup  []string
 	stopChan chan struct{}
 	doneChan chan struct{}
+
+	// tickSignal is a channel that, if non-nil and available, will be written
+	// to to signal that a tick has occurred. its sole purpose is for testing
+	// code, to verify that take cleanup attempts are happening when they
+	// should be.
+	tickSignal chan struct{}
```

Comment:
Can we instead use the side-effects caused by tick() to test ?

Reply:
not cleanly, as far as i could tell, unless you have a better idea?

additionally, it would make the test fragile to changes in the implementation of `tick`.

Reply:
bringing in a conversation @anshulpundir and i had in meatspace:

i've chosen to add this code for the purpose of determining affirmatively or negatively that `tick` has or _has not_ been called. Relying on side effects can tell that tick _has_ been called, but cannot prove definitively that it has _not_ been called.